### PR TITLE
feature: add support to compile this module as a nginx dynamic module

### PR DESCRIPTION
**Pull request by nettoclaudio**

### CODE REVIEW

File: config [hunk 1]
```diff
@@ -1,3 +1,12 @@
 ngx_addon_name=ngx_http_lua_ssl_module
-HTTP_MODULES="$HTTP_MODULES ngx_http_lua_ssl_module"
-NGX_ADDON_SRCS="$NGX_ADDON_SRCS $ngx_addon_dir/src/ngx_http_lua_ssl_module.c"
+
+if test -n "$ngx_module_link"; then
+  ngx_module_type=HTTP
+  ngx_module_name=ngx_http_lua_ssl_module
+  ngx_module_srcs="$ngx_addon_dir/src/ngx_http_lua_ssl_module.c"
```
Comment (spacewander) [inline config]: Can you put the common part (the module name, the module files) into variables to avoid repeating between two kinds of configure?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: config (hunk 1 above)
Comment (nettoclaudio) [reply config]: Sure! I'm changing here...

**Review by spacewander — APPROVED** [review]

**Review by rainingmaster — APPROVED** [review]
LGTM

**Comment by spacewander** [comment]
@nettoclaudio 
Merged, thanks!

**Pull request merged**